util/compression.h

```diff
@@ -46,6 +46,75 @@
 
 namespace rocksdb {
 
+// Data associated with compression dictionary, e.g., to avoid repetitive
+// preprocessing work.
+struct DictContext {
+#if defined(ZSTD) && ZSTD_VERSION_NUMBER >= 700  // v0.7.0+
+  // we cannot use unique_ptr as ZSTD_CDict is an incomplete type so its
+  // destructor is unknown. That's unfortunate because it means we have to
+  // implement move semantics ourselves.
+  ZSTD_CDict* zstd_cdict = nullptr;
+#endif
+  Slice raw_dict_bytes;
+
+  DictContext() {}
+
+  DictContext(DictContext&& other) {
+    *this = std::move(other);
+  }
+
+  DictContext& operator=(DictContext&& other) {
```

Comment:
Where do we need to make a copy?

Reply:
Oh it's move-assignment, which is just used by the move-constructor above. Now that I rethink it, we could probably just call `Init` in the `Rep` constructor and never have to move the `DictContext` anywhere...

Reply:
My choice would be to explicitly invoke default __Ctor in the init for the move __ctor 
DictContext(DictContext&& other) : DictContext()